dev-util/bear/bear-2.4.2.recipe

```diff
@@ -0,0 +1,58 @@
+SUMMARY="Automatically generate compilation database for Clang tooling"
+DESCRIPTION="Bear records the flags passed to the compiler for each translation unit and \
+stores them in a JSON file. This file can be used by Clang's tooling interface \
+and programs like clang-check to process a translation unit.\
+\
+cmake supports the generation of JSON compilation databases out of the box. \
+For any other build system that does not support this, Bear can be used \
+instead to intercept the invocation of the compiler."
+HOMEPAGE="https://github.com/rizsotto/Bear"
+COPYRIGHT="2012-2019 by László Nagy"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/rizsotto/Bear/archive/$portVersion.tar.gz"
+CHECKSUM_SHA256="e80c0d622a8192a1ec0c0efa139e5767c6c4b1defe1c75fc99cf680c6d1816c0"
+SOURCE_DIR="Bear-$portVersion"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64 ?arm"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	bear$secondaryArchSuffix = $portVersion compat >= 3
+	cmd:bear = $portVersion compat >= 3
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	cmd:python
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:cmake
+	cmd:g++
+	cmd:python
+	"
+
+BUILD()
+{
+	cmake . -DCMAKE_INSTALL_PREFIX=$prefix \
+		-DCMAKE_BUILD_TYPE=Release
+	make $jobArgs -C build
+}
+
+INSTALL()
+{
+	make install
+
+	mkdir -vp $manDir
+	cp -vrd $prefix/share/man/man1 $manDir
+
+	mkdir -p $dataDir
+	mv -v $prefix/share $dataDir
+}
```

Comment:
Could you add a TEST() section for "make check" ?

Reply:
Thanks, that's a good idea.

It's a little tricky because the tests in this project use the [lit tool from LLVM](https://github.com/llvm-mirror/llvm/tree/master/utils/lit) which isn't packaged currently. But I think it's worth it to set this up.

Right now I'm debating whether or not to update the `llvm` package to also provide the `lit` command, which I think is the most straight forward way to go.

Another alternative that feels a lot grosser is to just have the bear build script create a Python virtualenv and install `lit` from pypi, but I'd rather not go that route.

I'll look into setting that up. This PR may end up depending on a llvm PR.

Reply:
OK. We can also merge this PR and open an issue for the tests. A LLVM subpackage would be indeed the way to go.

Reply:
added an issue #4544